src/js/polyfills/console.js

```diff
@@ -1,258 +1,316 @@
+                    } else if (ri === 1) { // only draw the middle line after the header
+                        drawHorizLine(tableChars.leftMiddle, tableChars.rightMiddle+'\n', tableChars.rowMiddle);
+                    }
 
-        const objectValues = {};
-        const indexKeys = [];
-        const values = [];
-
-        const stringifyValue = value => format(value);
-        const toTable = (header, body) => print(cliTable(header, body));
-        const createColumn = (value, shift) => [
-            ...(shift ? [ ...new Array(shift) ].map(() => '') : []),
-            stringifyValue(value)
-        ];
-
-        let resultData;
-        const isSet = data instanceof Set;
-        const isMap = data instanceof Map;
-        const valuesKey = 'Values';
-        const indexKey = isSet || isMap ? '(iteration index)' : '(index)';
-
-        if (data instanceof Set) {
-            resultData = [ ...data ];
-        } else if (data instanceof Map) {
-            let idx = 0;
-
-            resultData = {};
-
-            data.forEach(
-                (v, k) => {
-                    resultData[idx] = { Key: k, Values: v };
-                    idx++;
+                    drawRow(rows[ri]);
                 }
-            );
-        } else {
-            resultData = data;
-        }
 
-        Object.keys(resultData).forEach(
-            (k, idx) => {
-                const value = resultData[k];
+                drawHorizLine(tableChars.bottomLeft, tableChars.bottomRight, tableChars.bottomMiddle);
 
-                if (value !== null && typeof value === 'object') {
-                    Object.entries(value).forEach(
-                        ([ k, v ]) => {
-                            if (properties && !properties.includes(k)) {
-                                return;
-                            }
+                return str;
+            }
 
-                            if (objectValues[k]) {
-                                objectValues[k].push(stringifyValue(v));
-                            } else {
-                                objectValues[k] = createColumn(v, idx);
-                            }
-                        }
-                    );
+            _printer('table', renderTable(rows, cols));
+        },
+        trace(...data) {
+            const stack = (new Error()).stack.trim().split('\n').slice(1).join('\n');
+
+            _printer('trace', 'Trace: ' + formatter(...data) + '\n' + stack);
+        },
+        dir(item, options) {
+            _printer('dir', inspect(item), options);
+        },
+        dirxml(...data) {
+            logger('dirxml', data.map(d=>inspect(d)));
+        },
+
+        // Counting
+        count(label = 'default') {
+            let count = countMap.get(label) ?? 0;
+
+            count++;
+            countMap.set(label, count);
+            _printer('count', label + ': ' + count);
+        },
+        countReset(label = 'default') {
+            if (!countMap.delete(label)) {
```

Comment:
According to the spec a reset sets it to 0, it doesn't delete it. Is trhere a practical difference?

Reply:
Yep, there is. When it is set to 0 you won't get a warning the next time when calling `.countReset()` again.

Though this is a weird case and nodejs does it the way I do. And I think it much better otherwise you never could clean up a counter and it would leak stuff.

```
> console.count('a')
a: 1
undefined
> console.countReset('a')
undefined
> console.countReset('a')
undefined
> (node:18291) Warning: Count for 'a' does not exist
(Use `node --trace-warnings ...` to show where the warning was created)
```

Reply:
Very subtle, let's keep it as is not to have an ever growing map indeed.